dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java

```diff
@@ -3829,6 +3829,55 @@ public void createWorkspaceItemFromExternalSourcesUnauthorized() throws Exceptio
                    .andExpect(status().isUnauthorized());
     }
 
+    @Test
+    public void createWorkspaceItemFromExternalSourcesNonAdminWithPermission() throws Exception {
+        //We turn off the authorization system in order to create the structure as defined below
+        context.turnOffAuthorisationSystem();
+        //** GIVEN **
+        //1. A community-collection structure with one parent community with sub-community and two collections.
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+        Community child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                                           .withName("Sub Community")
+                                           .build();
+        Collection col1 = CollectionBuilder.createCollection(context, child1).withName("Collection 1")
+                .withSubmitterGroup(eperson).build();
+
+        context.restoreAuthSystemState();
+
+        Integer workspaceItemId = null;
+
+        ObjectMapper mapper = new ObjectMapper();
+        String token = getAuthToken(eperson.getEmail(), password);
+        MvcResult mvcResult = getClient(token).perform(post("/api/submission/workspaceitems?owningCollection="
+                                          + col1.getID().toString())
+                                     .contentType(parseMediaType(
+                                         TEXT_URI_LIST_VALUE))
+                                     .content("https://localhost:8080/server/api/integration/externalsources/" +
+                                                  "mock/entryValues/one"))
+                                    .andExpect(status().isCreated()).andReturn();
+
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        workspaceItemId = (Integer) map.get("id");
+        String itemUuidString = String.valueOf(((Map) ((Map) map.get("_embedded")).get("item")).get("uuid"));
```

Comment:
@alemarte : This entire section of code can be simplified if you'd just use `andDo()`, similar to other `post` calls in this same IT class, e.g. https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java#L837-L842

The code you have here will "work", but it's much slower as you are loading the entire result into a Map in order to just get the `id` returned.  Instead, you should be able to do something like
```
AtomicReference<Integer> idRef = new AtomicReference<>();
getClient(token).perform(post("/api/submission/workspaceitems")
                    .param("owningCollection", col1.getID().toString())
                    .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
                    .content("https://localhost:8080/server/api/integration/externalsources/" +
                                                  "mock/entryValues/one"))
                    .andExpect(status().isCreated())
                    .andExcept(jsonPath("$._embedded.collection.id", is(col1.getID().toString())))
                    .andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "$.id")));
workspaceItemId = idRef.get();
```

We also should be *cleaning up* in this method, by having a `try()` / `finally`, with `WorkspaceItemBuilder.deleteWorkspaceItem(idRef.get());` in the `finally` block.

This change in code does mean that you won't be able to get the `itemUuidString`, but I don't think that's necessary for this test.  All you need to ensure is that the WorkspaceItem is created in the correct collection, having *metadata* which matches the mock external source.

Reply:
Thanks for your review @tdonohue. All the suggested improvements have been committed.